src/lib/vip-import-validate-files.js

```diff
@@ -0,0 +1,366 @@
+import { promisify } from 'util';
+
+// Promisify to use async/await
+const readDir = promisify( fs.readdir );
+
+// Accepted media file extensions
+export const acceptedExtensions = [
+	'jpg', 'jpeg', 'jpe',
+	'gif',
+	'png',
+	'bmp',
+	'tiff', 'tif',
+	'ico',
+	'asf',
+	'asx',
+	'wmv', 'wmx', 'wm',
+	'avi',
+	'divx',
+	'mov',
+	'qt',
+	'mpeg', 'mpg', 'mpe', 'mp4', 'm4v',
+	'ogv',
+	'webm',
+	'mkv',
+	'3gp', '3gpp', '3g2', '3gp2',
+	'txt',
+	'asc',
+	'c', 'cc', 'h',
+	'srt',
+	'csv', 'tsv',
+	'ics',
+	'rtx',
+	'css',
+	'vtt',
+	'dfxp',
+	'mp3',
+	'm4a', 'm4b',
+	'ra',
+	'ram',
+	'wav',
+	'ogg',
+	'oga',
+	'mid', 'midi',
+	'wma',
+	'wax',
+	'mka',
+	'rtf',
+	'js',
+	'pdf',
+	'class',
+	'psd',
+	'xcf',
+	'doc',
+	'pot',
+	'pps',
+	'ppt',
+	'wri',
+	'xla', 'xls', 'xlt', 'xlw',
+	'mdb', 'mpp',
+	'docx', 'docm', 'dotx', 'dotm',
+	'xlsx', 'xlsm', 'xlsb', 'xltx', 'xltm', 'xlam',
+	'pptx', 'pptm', 'ppsx', 'ppsm', 'potx', 'potm', 'ppam',
+	'sldx', 'sldm',
+	'onetoc', ' onetoc2', 'onetmp', 'onepkg', 'oxps',
+	'xps',
+	'odt', 'odp', 'ods', 'odg', 'odc', 'odb', 'odf',
+	'wp', 'wpd',
+	'key', 'numbers', 'pages',
+];
+
+/** Recommendations
+ *
+ * Recommend alternatives to invalid folders or files
+ */
+
+// Recommend the WordPress year/month file structure for media files
+const recommendedFileStructure = () => {
+	console.log(
+		chalk.underline( 'We recommend the WordPress default folder structure for your media files: \n\n' ) +
+		chalk.underline( 'Single sites:' ) +
+		chalk.yellow( ' `uploads/year/month/image.png` \n' ) +
+		' e.g.-' + chalk.yellow( '`uploads/2020/06/image.png` \n' ) +
+		chalk.underline( 'Multisites:' ) +
+		chalk.cyan( ' `uploads/sites/siteID/year/month/image.png` \n' ) +
+		' e.g.-' + chalk.cyan( '`uploads/sites/5/2020/06/images.png` \n' )
+	);
+	console.log( '------------------------------------------------------------' );
+	console.log();
+};
+
+// Recommend accepted file types
+const recommendAcceptableFileTypes = () => {
+	console.log(
+		'Accepted file types: \n\n' +
+		chalk.magenta( `${ acceptedExtensions }` )
+	);
+	console.log();
+};
+
+// Accepted file name characters
+const recommendAcceptableFileNames = () => {
+	const prohibitedCharacters = '+ & # % = \' \" \ < > : ; , / ? $ * | ` ! { }';
+	const acceptedCharacters = 'Non-English characters, spaces, ( ) [ ] ~';
+
+	console.log(
+		'The following characters are allowed in file names:\n' +
+		chalk.green( `${ acceptedCharacters }` ) + '\n\n' +
+		'The following characters are prohibited in file names:\n' +
+		chalk.red( `${ prohibitedCharacters }` )
+	);
+	console.log();
+};
+
+/** Nested Directory Search
+ *
+ * Use recursion to identify the nested tree structure of the folders
+ *
+ * @param {string} directory Root directory, or the current directory
+ */
+export const findNestedDirectories = async directory => {
+	let dir, nestedDir;
+
+	try {
+		// Read what's inside the current directory
+		dir = await readDir( directory );
+		nestedDir = dir[ 0 ];
```

Comment:
Can we define `0` as a constant or symbol for reference?

Reply:
Changed in f301178!